extension/src/SymbolReference/SymbolReferenceReader.ts

```diff
@@ -0,0 +1,263 @@
+    obj.generatedFromSymbol = true;
+    page.Properties?.forEach(prop => {
+        addProperty(prop, obj);
+    });
+    page.Controls?.forEach(control => {
+        addControl(control, obj);
+    });
+    return obj;
+}
+
+function addControl(control: ControlDefinition, parent: ALControl) {
+    let alControl: ALControl | undefined;
+    if (control.Kind === ControlKind.Field) {
+        const sourceExpr = control.Properties.filter(prop => prop.Name === 'SourceExpression')[0].Value;
+        alControl = new ALPageField(ALControlType.PageField, control.Name, sourceExpr);
+    } else if (control.Kind === ControlKind.Part) {
+        let value = control.RelatedPagePartId?.Name;
+        if (!value || value === '') {
+            value = control.RelatedPagePartId?.Id?.toString();
+        }
+        alControl = new ALPagePart(ALControlType.Part, control.Name, value || '');
+    } else {
+        let newAlControlType: ALControlType = ALControlType.None;
+        switch (control.Kind) {
+            case ControlKind.Area:
+                newAlControlType = ALControlType.Area;
+                break;
+            case ControlKind.CueGroup:
+                newAlControlType = ALControlType.CueGroup;
+                break;
+            case ControlKind.Group:
+                newAlControlType = ALControlType.Group;
+                break;
+            case ControlKind.Repeater:
+                newAlControlType = ALControlType.Repeater;
+                break;
+        }
+        if (newAlControlType !== ALControlType.None) {
+            alControl = new ALControl(newAlControlType, control.Name);
+        }
+    }
+    if (alControl !== undefined) {
+        control.Properties?.forEach(prop => {
+            if (alControl !== undefined) {
+                addProperty(prop, alControl);
+            }
+        });
+        alControl.parent = parent;
+        parent.controls.push(alControl);
+        control.Controls?.forEach(c => {
+            if (alControl !== undefined) {
+                addControl(c, alControl);
+            }
+        });
+    }
+}
+
+function addProperty(prop: SymbolProperty, obj: ALControl) {
+    let type = MultiLanguageTypeMap.get(prop.Name.toLowerCase());
+    if (type) {
+        let mlProp = new MultiLanguageObject(obj, type, prop.Name);
+        mlProp.text = prop.Value;
+        obj.multiLanguageObjects.push(mlProp);
+
+    } else if (ALPropertyTypeMap.has(prop.Name.toLowerCase())) {
+        obj.properties.push(new ALProperty(obj, 0, prop.Name, prop.Value));
+    }
+}
+
+
+export function getAppIdentifiersFromFilename(filePath: string) {
+    let fileName = path.basename(filePath);
+    const ext = path.extname(filePath);
+    fileName = fileName.substr(0, fileName.length - ext.length);
+    const appParts = fileName.split('_');
+    const name = appParts[1];
+    const publisher = appParts[0];
+    const version = appParts[2];
+    return { name, publisher, version };
```

Comment:
This suggestion is mainly for the return statement. TS handles the return type either way, but I guess we'd like to specify it? 
```suggestion
export function getAppIdentifiersFromFilename(filePath: string): { name: string, publisher: string, version: string } {
    let fileName = path.basename(filePath);
    const ext = path.extname(filePath);
    fileName = fileName.substr(0, fileName.length - ext.length);
    const appParts = fileName.split('_');
   return {
        name: appParts[1],
        publisher: appParts[0],
        version: appParts[2]
    };
```

Reply:
agreed. I saw this when I enabled "@typescript-eslint/explicit-function-return-type"